mcs/class/System/System.Net/HttpListenerRequest.cs

```diff
@@ -403,7 +403,7 @@ internal bool FlushInput ()
 		}
 
 		public bool IsLocal {
-			get { return IPAddress.IsLoopback (RemoteEndPoint.Address); }
+			get { return LocalEndPoint.Address.Equals (RemoteEndPoint.Address); }
```

Comment:
Reference source is doing it this way.
http://referencesource.microsoft.com/#System/net/System/Net/HttpListenerRequest.cs,458

When connecting to an endpoint the remote endpoint is specified but the local endpoint is not. I don't know if there is a way of connecting to a local endpoint through a different local endpoint.